exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/types/BinaryColumnTypeTests.kt

```diff
@@ -0,0 +1,41 @@
+package org.jetbrains.exposed.sql.tests.shared.types
+
+import org.jetbrains.exposed.dao.id.IntIdTable
+import org.jetbrains.exposed.sql.SchemaUtils
+import org.jetbrains.exposed.sql.binaryLiteral
+import org.jetbrains.exposed.sql.insertAndGetId
+import org.jetbrains.exposed.sql.selectAll
+import org.jetbrains.exposed.sql.tests.DatabaseTestsBase
+import org.jetbrains.exposed.sql.tests.shared.assertEqualLists
+import org.junit.Test
+
+class BinaryColumnTypeTests : DatabaseTestsBase() {
+    @Test
+    fun testBinaryColumn() {
+        val defaultValue = "default-value".toByteArray()
+        val tester = object : IntIdTable("testBinaryColumn") {
+            val bin = binary("bin", 256).default(defaultValue)
+        }
+
+        withTables(tester) {
+            assertEqualLists(emptyList(), SchemaUtils.statementsRequiredToActualizeScheme(tester))
+
+            val id1 = tester.insertAndGetId { }
+            val row1 = tester.selectAll().where { tester.id eq id1 }.first()
+            val binValue1 = row1[tester.bin]
+            assertEqualLists(defaultValue.toList(), binValue1.toList())
+
+            val literalValue = "literal-value".toByteArray()
+            val id2 = tester.insertAndGetId { it[tester.bin] = binaryLiteral(literalValue) }
```

Comment:
Literals are more meant to be used in query building, in WHERE clauses for example, or as column defaults, or where expressions are needed in functions. It might be good to at least test it it 1 of those actual use cases.

Reply:
TODO fix with [EXPOSED-433](https://youtrack.jetbrains.com/issue/EXPOSED-433) Check all the literals for default/insert/read operations